autotraining.lua

```diff
@@ -0,0 +1,258 @@
+-- Based on the original code by RNGStrategist (who also got some help from Uncle Danny)
+--@ enable = true
+--@ module = true
+
+local repeatUtil = require('repeat-util')
+local utils=require('utils')
+local dlg = require('gui.dialogs')
+
+validArgs = utils.invert({
+    't',
+    'n'
+})
+
+local args = utils.processArgs({...}, validArgs)
+local GLOBAL_KEY  = "autotraining"
+local ignore_flag = df.unit_labor['DISSECT_FISH']
+local need_id = df.need_type['MartialTraining']
+local ignore_count = 0
+
+local function get_default_state()
+    return {
+        enabled=false,
+        threshold=-5000,
+        squadname='Gym'
+    }
+end
+
+state = state or get_default_state()
+
+function isEnabled()
+    return state.enabled
+end
+
+-- Save any configurations in the save data
+local function persist_state()
+    dfhack.persistent.saveSiteData(GLOBAL_KEY, state)
+end
+
+dfhack.onStateChange[GLOBAL_KEY] = function(sc)
+    -- the state changed, is a map loaded and is that map in fort mode?
+    if sc ~= SC_MAP_LOADED or df.global.gamemode ~= df.game_mode.DWARF then
+        -- no its isnt, so bail
+        return
+    end
+    -- yes it was, so:
+
+    -- retrieve state saved in game. merge with default state so config
+    -- saved from previous versions can pick up newer defaults.
+    state = get_default_state()
+    utils.assign(state, dfhack.persistent.getSiteData(GLOBAL_KEY, state))
+    if ( state.enabled ) then
+        start()
+    else
+        stop()
+    end
+    -- start can change the enabled state if the squad cant be found
+    if state.enabled then
+        dfhack.print(GLOBAL_KEY .." was persisted with the following data:\nThreshold: ".. state.threshold .. ' | Squad name: '..state.squadname ..'\n')
+    end
+    persist_state()
+end
+
+
+--######
+--Functions
+--######
+function getAllCititzen()
```

Comment:
In order to avoid confusions with `getCitizens`, I would rename this function into something like `getTrainingCandidates`.